src/rubicon/objc/api.py

```diff
@@ -893,15 +892,27 @@ class or a metaclass, an instance of :class:`ObjCClass` or
                 # it's the correct class.
                 #
                 # Refs #249.
+                #
+                # We also need to evict the cache when the instance *class*
+                # doesn't match. This can happen when an ObjCInstance subclass
+                # (like ObjCMutableDictionary) has been used - an ObjCInstance
+                # will respond to the same selectors as ObjCMutableDictionary,
+                # but it won't have the helper methods (like __setitem__) that
+                # make the wrapper useful.
```

Comment:
I find this explanation bit hard to understand, especially the 1st sentence "when the instance *class* doesn't match". Are we talking about Python or ObjC classes here? I gather from the remaining text and the implementation that you mean Python classes. 

I'll leave a suggested rewrite, but feel free to reject or adapt. I understand that 'easy to understand' is very subjective :)

```
We also evict the cached instance when its Python type does not
match what we want to use to represent the given object in Python.
For example, we want to represent a NSDictionary instance as ObjCDictInstance to add Python dict API methods (like __setitem__). If we find a ObjCInstance instead in the cache, we evict it.
```

Reply:
Agreed that's a better explanation of this particular implementation; what I'm trying to work out (in response to @mhsmith's queries) is whether there is a better way to frame the original cache hit based on the actual ObjC class data. It feels like there should be... but I've been going mad trying to work out what that formulation would be. All the obvious stuff seems to have weird edge case failures.